### PR TITLE
Use 64 bits for int/float

### PR DESCRIPTION
**Pull request by terahxluna**
Closes https://github.com/Keats/tera/issues/79

Saw your your call for participation in This Week in Rust 159, here's my contribution =)

### CODE REVIEW

**Comment by Keats** [comment]
Thanks for that! 
i just pushed a big change to `0.5` branch so it needs to be rebased.

Would you be able to add a test in `render.rs` for math?

Something like:
```rust
#[test]
fn test_can_do_math() {
        let tests = vec![
            ("{{ 1 + 1 }}", "2"),
            ("{{ 1 + 1.1 }}", "2.1"),
            ("{{ 10 / 5 }}", "2"),
            // etc
        ];
        let mut tera = Tera::default();
        for (input, expected) in tests {
            tera.add_template("hey", input);
            assert_eq!(tera.render("hey", Context::new()), expected);
        }
}
```
Basically try to break the rendering using whatevet maths available

@SergioBenitez also left a comment in `render.rs` that applies here I believe:
```rust
// Since Tera only support 32 bit integers, this
// actually preserves all of the precision. If Tera
// switches to 64-bit values, use std::f32::EPSILON
// to get an approximation as before.
lhs_val = Value::F64(lhs_val.as_f64().unwrap());
rhs_val = Value::F64(rhs_val.as_f64().unwrap());
```

**Comment by terahxluna** [comment]
I rebased it on v0.5 and added some math tests. I'll take a look at @SergioBenitez 's comment tonight.

File: src/render.rs [hunk 1]
```diff
@@ -638,8 +638,13 @@ mod tests {
...
     #[test]
     fn test_render_math() {
-        let result = render_template("This is {{ 2000 + 16 }}.", Context::new());
-        assert_eq!(result.unwrap(), "This is 2016.".to_owned());
+        assert_eq!(render_template("{{ 1 + 1 }}", Context::new()).unwrap(), "2".to_owned());
+        assert_eq!(render_template("{{ 1 + 1.1 }}", Context::new()).unwrap(), "2.1".to_owned());
+        assert_eq!(render_template("{{ 3 - 1 }}", Context::new()).unwrap(), "2".to_owned());
+        assert_eq!(render_template("{{ 3 - 1.1 }}", Context::new()).unwrap(), "1.9".to_owned());
+        assert_eq!(render_template("{{ 2 * 5 }}", Context::new()).unwrap(), "10".to_owned());
+        assert_eq!(render_template("{{ 10 / 5 }}", Context::new()).unwrap(), "2".to_owned());
```
Comment (Keats) [inline src/render.rs]: can you add `int * float` and `int / float` as well please?

**Comment by Keats** [comment]
Looks good, thanks! Let me know when you are done with it

**Comment by Keats** [comment]
@terahxluna ping

**Comment by terahxluna** [comment]
Sorry for the delay.

I'm having some trouble to understand what would the use of std::f32::EPSILON do.

For what I understand, a conversion already takes place in lhs_val.as_f64(), so the loss of precision has already took place if the number is bigger than the fraction part of f64.

**Comment by Keats** [comment]
Hmm i'll have a look later then, thanks!

**Pull request merged**